### PR TITLE
Remove InitBy*. Added support for fill rate

### PR DESCRIPTION
**Pull request by ashtul**

### CODE REVIEW

**Comment by ashtul** [comment]
By popular demand, CMS.RESERVE got probability option.
CMS.INFO got capacity  and fill rate

File: docs/CountMinSketch_Commands.md [hunk 1]
```diff
@@ -1,50 +1,163 @@
-# RedisBloom Count Min Sketch Documentation
+# RedisBloom Count-Min Sketch Documentation
 
-## CMS.INITBYDIM
+## Create
 
-### Format:
+### CMS.RESERVE
 
+Initializes a Count-Min Sketch to the to accommodate requested capacity.
```
Comment (mnunberg) [inline docs/CountMinSketch_Commands.md]: "to the to"... fix plz

File: docs/CountMinSketch_Commands.md [hunk 2]
```diff
@@ -1,50 +1,163 @@
...
+CMS.RESERVE test 1000
 ```
 
-Initializes a CMS to the dimensions specified by width and depth.
+## Update
+
+### CMS.INCRBY
 
-## Parameters:
+Increases the count of item by increment. Multiple items can be increase with one call.
```
Comment (mnunberg) [inline docs/CountMinSketch_Commands.md]: increased

File: docs/CountMinSketch_Commands.md [hunk 3]
```diff
@@ -1,50 +1,163 @@
...
+### CMS.INCRBY
 
-## Parameters:
+Increases the count of item by increment. Multiple items can be increase with one call. 
 
-* **key**: Key under which the sketch is to be found.
-* **width**: Number of counters kept in each array.
-* **Depth**: Number of arrays.
+```sql
+CMS.INCRBY key item increment [item increment]
```
Comment (mnunberg) [inline docs/CountMinSketch_Commands.md]: `...`

File: docs/CountMinSketch_Commands.md [hunk 4]
```diff
@@ -1,50 +1,163 @@
...
+CMS.INCRBY test foo 10 bar 42
+```
 
-## CMS.INITBYPROB
+## Query
 
-### Format:
+### CMS.QUERY
 
+Increases the count of item by increment. Multiple items can be increase with one call.
```
Comment (mnunberg) [inline docs/CountMinSketch_Commands.md]: I don't think this is correct

File: src/rm_cms.c [hunk 5]
```diff
@@ -102,46 +102,42 @@ int CMSketch_Create(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
...
     }
     if (argc == 3) {
-        *width = size * DEFAULT_WIDTH;
+        *width = *size * DEFAULT_WIDTH;
         *depth = DEFAULT_DEPTH;
     } else {
-        if (RMUtil_ArgIndex("ERROR", argv, argc) != 3) {
-            INNER_ERROR("CMS: invalid ERROR");
+        if (RMUtil_ArgIndex("PROBABILITY", argv, argc) != 3) {
+            INNER_ERROR("CMS: PROBABILITY word not if 4th place");
```
Comment (mnunberg) [inline src/rm_cms.c]: just say "Missing probability (or bad arguments)"

File: src/rm_cms.c [hunk 6]
```diff
@@ -102,46 +102,42 @@ int CMSketch_Create(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
...
         }
-        if ((RedisModule_StringToDouble(argv[4], &overEst) != REDISMODULE_OK) || overEst <= 0 ||
-            overEst >= 1) {
-            INNER_ERROR("CMS: invalid overestimation value");
+        if ((RedisModule_StringToDouble(argv[4], &err) != REDISMODULE_OK) ||
```
Comment (mnunberg) [inline src/rm_cms.c]: If you used ArgCursor (in args.c/args.h, it would do this checks for you)

**Comment by ashtul** [comment]
@mnunberg requested changes implemented

**Comment by ashtul** [comment]
Revert to InitByDim and InitByProb

File: docs/CountMinSketch_Commands.md [hunk 7]
```diff
@@ -1,21 +1,55 @@
...
+```sql
+CMS.INITBYDIM test 2000 5
+```
+
+### CMS.INITBYPROB
+
+Initializes a Count-Min Sketch to accommodate requested capacity.
+
+```sql
+CMS.INITBYPROB key error probability
```
Comment (jeffreylovitz) [inline docs/CountMinSketch_Commands.md]: I get an arity error with this:
```
CMS.INITBYPROB test 0.001 0.01
(error) ERR wrong number of arguments for 'CMS.INITBYPROB' command
```

File: docs/CountMinSketch_Commands.md [hunk 8]
```diff
@@ -1,21 +1,55 @@
...
+O(1)
+
+### Return
+
+OK on success, error otherwise
 
-* **key**: Key under which the sketch is to be found.
-* **width**: Number of counters kept in each array.
-* **Depth**: Number of arrays.
+#### Reserve example
```
Comment (jeffreylovitz) [inline docs/CountMinSketch_Commands.md]: Not a `Reserve` command - you might just want to go with a generic `Example` header.

File: docs/CountMinSketch_Commands.md [hunk 9]
```diff
@@ -24,27 +58,132 @@ O(1)
 
 OK on success, error otherwise
 
-## CMS.INITBYPROB
+#### Reserve example
```
Comment (jeffreylovitz) [inline docs/CountMinSketch_Commands.md]: Not a `Reserve` command - you might just want to go with a generic `Example` header.

File: docs/CountMinSketch_Commands.md (hunk 7 above)
Comment (ashtul) [reply docs/CountMinSketch_Commands.md]: Fixed

File: docs/CountMinSketch_Commands.md (hunk 9 above)
Comment (ashtul) [reply docs/CountMinSketch_Commands.md]: Changed to generic example

File: src/rm_cms.c [hunk 10]
```diff
@@ -59,27 +62,23 @@ static int parseCreateArgs(RedisModuleCtx *ctx, RedisModuleString **argv, int ar
     } else {
         long long n = 0;
```
Comment (jeffreylovitz) [inline src/rm_cms.c]: Unused - delete if definitely unnecessary for `InitByProb`?

**Pull request merged**